workspaces/ui-v2/src/store/root.ts

```diff
@@ -0,0 +1,16 @@
+import { configureStore } from '@reduxjs/toolkit';
+import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
+
+import { reducer as endpointsReducer } from './endpoints';
+
+export const store = configureStore({
+  reducer: {
+    endpoints: endpointsReducer,
+  },
+});
+
+export type RootState = ReturnType<typeof store.getState>;
+export type AppDispatch = typeof store.dispatch;
+
+export const useAppDispatch = () => useDispatch<AppDispatch>();
+export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
```

Comment:
We use these custom dispatch / selectors such that our hooks have the appropriate redux types

Reply:
Ahh gotcha, makes sense!